src/plugins/email/testmod_email.c

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file
+ *
+ * @brief Tests for email plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <stdio.h>
+
+#define PLUGIN_NAME "email"
+#include "test_email.h"
+
+int main (int argc, char ** argv)
+{
+	printf ("EMAIL        TESTS\n");
+	printf ("==================\n\n");
+
+	init (argc, argv);
+	testEmailAll ();
```

Comment:
Nice, but I don't get why you put the implementation in the header file?

Reply:
I took src/plugins/ipaddr as a template.
It is done like this there.